### PR TITLE
fix(api, shared-data): Allow labware lids to be disposed in the trash bin

### PR DESCRIPTION
**Pull request by CaseyBatten**
# Overview
Covers PLAT-539

This allows the TC lid to be dropped in the trash bin at a slight offset. Originally we only allowed it to be dropped in the waste chute, but we need to let it be disposed of in some sanitary way in other disposal locations as well.

## Test Plan and Hands on Testing

- [x] Verify that we don't collide with labware in slot above the trashbin when dropping the lid
- [x] Verify that stack of lids can fit within trash bin (do we need to implement some kind of alternation like with tips? Is that safe?)

The following protocol should move a TC lids from a stack on the deck into a trash bin:
```
from typing import List
from opentrons.protocol_api import ProtocolContext, Labware

metadata = {"protocolName": "Opentrons Flex TC Lid Trash Test"}
requirements = {"robotType": "Flex", "apiLevel": "2.20"}


LID_COUNT = 5
LID_DEFINITION = "opentrons_tough_pcr_auto_sealing_lid"
LID_BOTTOM_DEFINITION = "opentrons_tough_pcr_auto_sealing_lid"

def run(protocol: ProtocolContext):
    # Tiprack to occupy space above test trashbin for collision checking
    tiprack = protocol.load_labware("opentrons_flex_96_tiprack_200ul", 'C3')

    trash = protocol.load_trash_bin('D3')

    lids: List[Labware] = [protocol.load_labware(LID_BOTTOM_DEFINITION, 'D2')]
    for i in range(LID_COUNT - 1):
        lids.append(lids[-1].load_labware(LID_DEFINITION))
    lids.reverse()  # NOTE: reversing to more easily loop through lids from top-to-bottom

    for lid in lids:
        protocol.move_labware(
            lid,
            trash,
            use_gripper=True,
        )
```
## Changelog

- Allow loaded trashes to be passed as a destination for move_labware commands
- Validated that when trashbins are a destination for labware, that the labware is exclusively a lid-type
- Raise analysis errors for unsupported labware trashing behavior, and errors for trashing lids that don't specifically have dedicated trash offsets (double restriction)

## Review requests
Do movements achieve what we want to unblock users without waste chutes? Do those analysis error messages make sense for when people try to do unsupported actions?

## Risk assessment
This currently allows any labware with the allowed role of "Lid" to be dropped in the trash bin. This means users could start trashing other lids in the future, so we may want to limit this via a labware quirk instead. On the other hand, giving users a sanitary way to dispose of any and all future lid types even if they don't own a waste chute may be useful. Thoughts?

### CODE REVIEW

File: api/src/opentrons/protocol_engine/commands/move_labware.py [hunk 1]
```diff
@@ -162,6 +169,37 @@ async def execute(self, params: MoveLabwareParams) -> _ExecuteReturn:  # noqa: C
                     y=0,
                     z=0,
                 )
+            elif fixture_validation.is_trash(area_name):
+                # When dropping labware in the trash bins we want to ensure they are lids
+                # and enforce a y-axis drop offset to ensure they fall within the trash bin
+                if labware_validation.validate_definition_is_lid(
+                    self._state_view.labware.get_definition(params.labwareId)
+                ):
+                    self._state_view.labware.get_labware_gripper_offsets
```
Comment (vegano1) [inline api/src/opentrons/protocol_engine/commands/move_labware.py]: what is this doing?

Comment (CaseyBatten) [reply api/src/opentrons/protocol_engine/commands/move_labware.py]: secondary validation that the labware we are about to move to the trash is infact a lid. We validate this once at the PAPI call, but we validate it here to catch incase someone tries to do this during a maintenance run or through other direct-engine command implementations.

File: api/src/opentrons/protocol_engine/commands/move_labware.py [hunk 2]
```diff
@@ -162,6 +169,37 @@ async def execute(self, params: MoveLabwareParams) -> _ExecuteReturn:  # noqa: C
...
+                            x=current_labware_definition.gripperOffsets[
+                                "lidDisposalOffsets"
+                            ].dropOffset.x,
+                            y=current_labware_definition.gripperOffsets[
+                                "lidDisposalOffsets"
+                            ].dropOffset.y,
+                            z=current_labware_definition.gripperOffsets[
+                                "lidDisposalOffsets"
+                            ].dropOffset.z,
+                        )
```
Comment (vegano1) [inline api/src/opentrons/protocol_engine/commands/move_labware.py]: Could change to

```
lid_disposable_offfets = current_labware_definition.gripperOffsets.get('lidDisposalOffsets')
if lis_disposable_offsets is not None:
    trash_lid_drop_offset = LabwareOffsetVector(**lid_disposable_offsets['dropOffset'])
else:
    raise ...
```

File: api/src/opentrons/protocol_engine/resources/fixture_validation.py [hunk 3]
```diff
@@ -29,7 +29,12 @@ def is_drop_tip_waste_chute(addressable_area_name: str) -> bool:
 
 def is_trash(addressable_area_name: str) -> bool:
     """Check if an addressable area is a trash bin."""
-    return addressable_area_name in {"movableTrash", "fixedTrash", "shortFixedTrash"}
+    return any(
```
Comment (vegano1) [inline api/src/opentrons/protocol_engine/resources/fixture_validation.py:32]: does not look like this is doing anything differently, if so let's revert.

**Review by vegano1 — APPROVED** [review]
Very nice,
I left a few comments
I tested on the Flex and everything works as expected.

File: api/src/opentrons/protocol_engine/commands/move_labware.py (hunk 1 above)
Comment (vegano1) [reply api/src/opentrons/protocol_engine/commands/move_labware.py]: yeah that i see, im talking specifically about
```
self._state_view.labware.get_labware_gripper_offsets
```

File: api/src/opentrons/protocol_engine/resources/fixture_validation.py (hunk 3 above)
Comment (CaseyBatten) [reply api/src/opentrons/protocol_engine/resources/fixture_validation.py:32]: Does produce different results unfortunately. The addressable area names for our trashes as moveableTrash + Deck slot name (like MoveableTrashD3). When looking for the full name of our addressable area in that list we get no results, but when looking for the individual substrings of that list within the name of our addressable area we get a positive result.

Comment (vegano1) [reply api/src/opentrons/protocol_engine/resources/fixture_validation.py:32]: oh i see, cool

File: api/src/opentrons/protocol_engine/commands/move_labware.py (hunk 1 above)
Comment (CaseyBatten) [reply api/src/opentrons/protocol_engine/commands/move_labware.py]: Actually just got that accidental variable call you highlighted on 178, thats unintentional good catch.

**Review by vegano1 — APPROVED** [review]

**Pull request merged**